models/domain.go

```diff
@@ -77,8 +77,7 @@ func (d *Domain) IsQueueable(db scanStore, list policyList) (bool, string, Scan)
 // PopulateFromScan updates a Domain's fields based on a scan of that domain.
 func (d *Domain) PopulateFromScan(scan Scan) {
 	// We should only trust MTA-STS info from a successful MTA-STS check.
-	if scan.Data.MTASTSResult != nil && scan.SupportsMTASTS() {
-		d.MTASTSMode = scan.Data.MTASTSResult.Mode
+	if d.MTASTSMode == "on" && scan.Data.MTASTSResult != nil && scan.SupportsMTASTS() {
```

Comment:
`scan.Data.MTASTSResult != nil` is buit into `scan.SupportsMTASTS()` so we can just

```suggestion
	if d.MTASTSMode == "on" && scan.SupportsMTASTS() {
```

:smiley_cat: